### DISCUSSION TITLE
quic支持剥离tls，只跑传输层部分吗，不带加密部分？

### DISCUSSION THREAD

**Discussion by flyingeagle23** [alibaba/xquic#335] [Q&A]
quic支持剥离tls，只跑传输层部分吗，不带加密部分？

**Comment by Kulsk** [comment]
目前是不太现实的，因为标准协议里边耦合了TLS加密握手。如果出于性能考虑，可以尝试xquic的no_crypto选项，但是只能针对0RTT、1RTT包不做加密，还是会尝试加密握手。

no_crypto选项在客户端尝试连接的时候带入，该参数是我们自己实现的，非官方标准，可以参考下：
![image](https://github.com/alibaba/xquic/assets/22413114/a4cf743b-f6ed-474a-bd7f-ffe3a8050ac5)

**Comment by hankai17** [comment]
可以看一下我的这个项目 
https://github.com/hankai17/quic-fiber
明文交互
